### PR TITLE
Replaced onDarkBackground in Link with darkBackground

### PR DESCRIPTION
**Pull request by robinpiets**

### CODE REVIEW

File: packages/asc-ui/src/components/TopTaskLink/TopTaskLinkStyle.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export const TopTaskTitle = styled(Paragraph)`
 `
 
 export default styled.a<{
-  onDarkBackground?: boolean
+  hasDarkBackground?: boolean
```
Comment (jonkoops) [inline packages/asc-ui/src/components/TopTaskLink/TopTaskLinkStyle.ts:23]: I don't think we can simply rename this prop as we need to still support the deprecated variant.

File: packages/asc-ui/src/components/Alert/Alert.stories.mdx [hunk 2]
```diff
@@ -92,14 +92,14 @@ export const Decorator = styled.div`
           Accusamus ad illo molestiae vel. Deleniti excepturi iusto neque
           quisquam!
         </Paragraph>
-        <Link onDarkBackground href="/" variant="with-chevron">
+        <Link href="/" variant="with-chevron" hasDarkBackground>
```
Comment (jonkoops) [inline packages/asc-ui/src/components/Alert/Alert.stories.mdx:95]: Perhaps we can just shorten this to `darkBackground`?

File: packages/asc-ui/src/components/TopTaskLink/TopTaskLinkStyle.ts (hunk 1 above)
Comment (robinpiets) [reply packages/asc-ui/src/components/TopTaskLink/TopTaskLinkStyle.ts]: Good catch.

Fixed all improvements 👍
